### PR TITLE
Support for Sail CI

### PR DESCRIPTION
**Pull request by jonsharratt**
We are a new CI  🎉 and would love to be a part of the ever growing list of CI tools available as part of your project.

**What was implemented:**

* Add mapping for Sail CI environment variables
* Add tests for mapping of Sail CI environment variables
* Add Sail CI to README
* Ensured that all ordering was in alphabetical order
* Followed commit message style based on commit history

Let me know if I have missed anything and I will update the pull request as per your guidance. Is a cool project idea 👍

### CODE REVIEW

File: lib/sailci.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// http://docs.sail.ci/documentation/environment-variables
+
+module.exports = {
+	detect({env}) {
+		return Boolean(env.SAILCI);
+	},
+	configuration({env}) {
+		return {
+			name: 'Sail CI',
+			service: 'sailci',
```
Comment (pvdlg) [inline lib/sailci.js]: For consistency reasons could you set the `service` to `sail` rather than `sailci`?

File: lib/sailci.js [hunk 2]
```diff
@@ -0,0 +1,17 @@
+// http://docs.sail.ci/documentation/environment-variables
```
Comment (pvdlg) [inline lib/sailci.js]: Can you rename this file `sail.js` instead of `sailci.js`?

File: test/sailci.test.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+import test from 'ava';
```
Comment (pvdlg) [inline test/sailci.test.js]: Similarly the file should be named `test/sail.test.js` rather than `test/sailci.test.js`.

File: README.md [hunk 4]
```diff
@@ -71,6 +71,7 @@ if (isCI) {
 | [Drone](http://readme.drone.io/0.5/usage/environment-reference)                                                |   `drone`   |   ✅    |    ✅     |    ✅    |     ❌      |    ✅     |   ✅   |    ❌     |  ✅   |   ✅    |   ✅    |   ❌    |
 | [Gitlab CI/CD](https://docs.gitlab.com/ce/ci/variables/README.html)                                            |  `gitlab`   |   ✅    |    ✅     |    ✅    |     ✅      |    ✅     |   ✅   |    ✅     |  ❌   |   ❌    |   ✅    |   ✅    |
 | [Jenkins](https://wiki.jenkins.io/display/JENKINS/Building+a+software+project)                                 |  `jenkins`  |   ✅    |    ✅     |    ✅    |     ✅      |    ✅     |   ❌   |    ❌     |  ❌   |   ❌    |   ✅    |   ✅    |
+| [Sail CI](http://docs.sail.ci/documentation/environment-variables)                                             |  `sailci`   |   ✅    |    ✅     |    ❌    |     ❌      |    ✅     |   ❌   |    ❌     |  ❌   |   ❌    |   ✅    |   ✅    |
```
Comment (pvdlg) [inline README.md]: Can you change the `service` to `sail` instead of `sailci`?

File: test/sailci.test.js [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+import sailci from '../lib/sailci';
+
+/* eslint-disable camelcase */
```
Comment (pvdlg) [inline test/sailci.test.js]: This eslint-disable shouldn't be necessary here as there is no lower camel-case props.

**Comment by pvdlg** [comment]
Thanks for the PR!

Do you intend to add to Sail CI environment variables to determine the build ID, the build URL, the PR number (if the build was triggered by a PR)?

This project was created mainly for semantic-release, and one the variable we use there is `isPr` as semantic-release skips PRs.
It's also quite convenient to have a Build ID and a URL so one can provide more useful logs to users.

**Comment by jonsharratt** [comment]
Hey @pvdlg - managed to get round to re-visit this and have updated the PR with latest changes we have made after discussing a few things incorporating your helpful insights and feedback with the team.

> Do you intend to add to Sail CI environment variables to determine the build ID, the build URL, the PR number (if the build was triggered by a PR)?

We have added the pull request number which sets set the `true` / `false` to determine when a build is a PR.

We don't currently have a requirement for a build id or build url as we stream our logs directly into GitHub via GitHub checks  (example https://github.com/sheepsteak/react-shallow-testutils/pull/35/checks). We will no doubt have a UI in the future to replicate functionality for other source control providers that do not support a feature such as GitHub checks.  It is not in the near future but will obviously raise a new PR when and if it happens.

Do let me know if I have missed anything incorporating changes required for this PR to be merged.

🎉 🍾

**Comment by pvdlg** [comment]
Thanks so much for updating the PR!
Can you just add a test when the `SAIL_PULL_REQUEST_NUMBER ` is set? Similarly to https://github.com/pvdlg/env-ci/blob/6f9321d7c8d13c118443ba5334fa23098f805211/test/travis.test.js#L32.

And one more question, when a Sail build is triggered by a PR, is `SAIL_COMMIT_BRANCH` contains the branch from which the PR originated (e.g. `feature/something`) or the target branch (e.g. `master`)?

I'm trying (in a future version of `env-ci`) to harmonize the behavior of the `branch` property. It seems CI services behave differently in that regard, some set their branch environment variable to the branch from which the PR originated, some to the target branch and some have a dedicated environment variables in case of a PR build to avoid confusion.

**Comment by jonsharratt** [comment]
Updated to replicate the testing pattern for `Push` and `PR`.

As for the  `SAIL_COMMIT_BRANCH` we looked at CircleCI and replicated their functionality and pass through in the format `pull/35`.

**Comment by pvdlg** [comment]
Thanks a lot !

> As for the SAIL_COMMIT_BRANCH we looked at CircleCI and replicated their functionality and pass through in the format pull/35.

Is there is way, in builds triggered by a PR, to know the branch from which the PR originated or the target branch? It's quite useful to know at least the target branch, in order to run things only for PR targeting a specific branch.

Travis, for example, in a PR triggered build, sets `TRAVIS_BRANCH` to the target branch and `TRAVIS_PULL_REQUEST_BRANCH` to the branch from which the PR originated.

For info, it seems most CIs allow to determine the PR target branch. The exceptions so far in my tests are Buddy (See https://forum.buddy.works/t/determine-pull-request-branch-with-environment-variable/911) and CircleCI (See https://discuss.circleci.com/t/create-a-circle-target-branch-envar/10022).

**Pull request merged**